libraries/engage/core/hooks/useWidgetConfig.js

```diff
@@ -1,46 +1,15 @@
-import defaultsDeep from 'lodash/defaultsDeep';
-import { themeConfig } from '@shopgate/pwa-common/helpers/config';
 import { useRoute } from './useRoute';
-import { usePageSettings } from './usePageSettings';
-import { getThemeSettings } from '../config/getThemeSettings';
+import { getWidgetConfig } from '../config/getWidgetConfig';
 
 /**
- * Retrieves the widget-configuration for a specific widget by its ID and an optional index.
- * If no index is given it will return the first found widget-configuration by widgetId.
+ * Retrieves the pure config for widget by its id and optional index on the current page.
+ * See documentation of `getWidgetConfig` for further details.
  *
  * @param {string} widgetId The id of the widget to look for, which must exist in the config.
- * @param {number|undefined} [index] The optional index of the widget.
+ * @param {number|undefined} [index=0] The optional index of the widget.
  * @returns {Object}
  */
-export function useWidgetConfig(widgetId, index) {
-  const { pattern } = useRoute();
-  const { pages } = themeConfig;
-  const { [widgetId]: pageSettings = {} } = usePageSettings();
-  const { [widgetId]: globalSettings = {} } = getThemeSettings();
-  const page = pages.find(element => element.pattern === pattern);
-
-  if (!page) {
-    return { settings: globalSettings };
-  }
-
-  const widget = [].concat(page.widgets).find((element, i) => {
-    if (index === undefined) {
-      return element.id === widgetId;
-    }
-
-    return element.id === widgetId && i === index;
-  });
-
-  if (!widget) {
-    return { settings: defaultsDeep(pageSettings, globalSettings) };
-  }
-
-  const { name, id, ...config } = widget;
-
-  const settings = defaultsDeep(config.settings, pageSettings, globalSettings);
-
-  return {
-    ...config,
-    settings,
-  };
+export function useWidgetConfig(widgetId, index = 0) {
+  const { pattern: pagePattern } = useRoute();
```

Comment:
This is not needed if the `useCurrentRoute()` helper is used inside `getPageConfig()` as mentioned above.

Reply:
See my previous comment.